packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -77,6 +91,14 @@ function isPathComplete(path: Fill[], targetInput: BigNumber): boolean {
     return input.gte(targetInput);
 }
 
+function clipFillAdjustedOutput(fill: Fill, remainingInput: BigNumber): BigNumber {
+    if (fill.input.lte(remainingInput)) {
+        return fill.adjustedOutput;
+    }
+    const penalty = fill.adjustedOutput.minus(fill.output);
+    return fill.output.times(remainingInput.div(fill.input)).plus(penalty);
```

Comment:
This is negative right? If so it would be clearer if it was: 

```suggestion
    const penalty = fill.output.minus(fill.adjustedOutput);
    return fill.output.times(remainingInput.div(fill.input)).minus(penalty);
```

Reply:
Nah, it can be positive for buys.